### PR TITLE
[SFSRAP-17] Fixed the generated property names to the format expected…

### PR DESCRIPTION
**Pull request by duncanpMS**
… by the C# plugin

### CODE REVIEW

File: PluginGenerator/PluginBuilder.cs [hunk 1]
```diff
@@ -264,6 +264,18 @@ protected void SetSourceFileReplacement(string key, string value)
...
+            string pluginKey;
+            this.properties.TryGetValue(WellKnownPluginProperties.Key, out pluginKey);
+            if (pluginKey != null)
+            {
+                // By convention this is all lower-case
+                pluginKey = pluginKey.ToLowerInvariant();
+            }
+            return pluginKey;
+        }
+
```
Comment (duncanpMS) [inline PluginGenerator/PluginBuilder.cs]: Made protected so builder sub-classes can access it.

File: PluginGenerator/PluginDefinition.cs [hunk 2]
```diff
@@ -27,7 +28,19 @@ public PluginManifest()
...
+            get { return this.pluginKey; }
+            set
+            {
+                this.pluginKey = value;
+                if (this.pluginKey != null)
+                {
+                    this.pluginKey = this.pluginKey.ToLowerInvariant();
+                }
+            }
+        }
```
Comment (duncanpMS) [inline PluginGenerator/PluginDefinition.cs]: Force the manifest to return a correctly-cased result too.

File: PluginGenerator/Resources/Core/Plugin.java [hunk 3]
```diff
@@ -20,6 +20,10 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(Plugin.class);
 
+  protected static final String KEY = "[PLUGIN_KEY]";
+
+  protected static final String NAME = "[PLUGIN_NAME]";
+
```
Comment (duncanpMS) [inline PluginGenerator/Resources/Core/Plugin.java]: Added some new protected properties to the core plugin Java class so they can be referenced by subclasses.
Previously the subclasses were directly referring to the source code substitution tokens (e.g. _[XXX]_ ) which was more error-prone and harder to debug. Now we'd get compilation errors which are easier to fix.

File: PluginGenerator/Resources/Rules/PluginRulesDefinition.java [hunk 4]
```diff
@@ -22,21 +22,19 @@
   private static final Logger LOG = LoggerFactory.getLogger(PluginRulesDefinition.class);
 
   // Unique to the plugin: used when naming the repository and the properties
-  protected static final String KEY = "[PLUGIN_KEY]";
+  protected static final String REPOSITORY_ID = "[RULE_REPOSITORY_ID]";
   
-  protected static final String NAME = "[PLUGIN_NAME]";
```
Comment (duncanpMS) [inline PluginGenerator/Resources/Rules/PluginRulesDefinition.java]: Now exposed by the core plugin class.

File: PluginGenerator/Resources/Rules/PluginRulesDefinition.java [hunk 5]
```diff
@@ -22,21 +22,19 @@
   private static final Logger LOG = LoggerFactory.getLogger(PluginRulesDefinition.class);
 
   // Unique to the plugin: used when naming the repository and the properties
-  protected static final String KEY = "[PLUGIN_KEY]";
+  protected static final String REPOSITORY_ID = "[RULE_REPOSITORY_ID]";
```
Comment (duncanpMS) [inline PluginGenerator/Resources/Rules/PluginRulesDefinition.java]: The [PLUGIN_KEY] symbol was being used for a number of logically different purposes. I've introduced this new symbol ([RULE_REPOSITORY_KEY]) to make it clearer what the value represents.

File: Tests/IntegrationTests/Roslyn/RoslynGenTests.cs [hunk 6]
```diff
@@ -34,7 +34,7 @@ public void RoslynPlugin_GenerateForValidAnalyzer_Succeeds()
             // Create a valid analyzer package
             ExampleAnalyzer1.CSharpAnalyzer analyzer = new ExampleAnalyzer1.CSharpAnalyzer();
 
-            string packageId = "analyzer1.pkgid1";
+            string packageId = "Analyzer1.Pkgid1"; // package id is not all lowercase
```
Comment (duncanpMS) [inline Tests/IntegrationTests/Roslyn/RoslynGenTests.cs]: Fixed up the test to check casing is correct.

**Comment by bgavrilMS** [comment]
LGTM

**Pull request merged**